src/components/Classes/components/JoinClassModal.js

```diff
@@ -1,89 +1,107 @@
+
+  const closeModal = () => {
+    if (props.onClose && {}.toString.call(props.onClose) === '[object Function]') {
+      props.onClose();
     }
 
-    this.setState({
-      wid: "",
-      error: "",
-      disableSubmit: false,
-    });
+    setWid('');
+    setError('');
+    // setDisableSubmit(false)
+
+    // this.setState({
+    //   wid: "",
+    //   error: "",
+    //   disableSubmit: false,
+    // });
   };
 
-  badInput = () => {
-    if (!this.state.wid) {
-      this.setState({
-        error: "Please enter the class code. You should get this from your instructor.",
-      });
+  const badInput = () => {
+    if (!wid) {
+      setError('Please enter the class code. You should get this from your instructor.');
+      // this.setState({
+      //   error: "Please enter the class code. You should get this from your instructor.",
+      // });
       return true;
     }
     // TODO: Check if class code input is formatted correctly
 
     return false;
   };
 
-  onSubmit = async (e) => {
+  const onSubmit = async (e) => {
     e.preventDefault();
 
-    if (this.badInput()) {
+    if (badInput()) {
       return;
     }
 
-    let data = {
-      uid: this.props.uid,
+    const data = {
+      uid: props.uid,
       // Todo: swap this line back in when BE is fixed
       // wid: this.state.wid,
-      cid: this.state.wid,
+      cid: wid,
     };
 
     try {
       fetch
         .joinClass(data)
         .then((res) => {
           if (!res.ok) {
-            if (res.status == 404) {
-              throw "404";
+            if (res.status === 404) {
+              throw new Error('404');
             } else {
               throw new Error(`Error: Response code is ${res.status}`);
             }
           }
           return res.classData;
         })
         .then((json) => {
-          this.props.addStudentClass(json.cid, json || {});
-          this.setState({ redirect: true });
-          this.closeModal();
+          props.addStudentClass(json.cid, json || {});
+          // this.setState({ redirect: true });
+          setRedirect(true);
+          closeModal();
         })
         .catch((err) => {
-          this.setState({
-            disableSubmit: false,
-            error: (err === "404") ?
-              "We couldn't find that class. Please try again!" :
-              "There was a problem joining the class, please try again!",
-          });
-          console.log(err);
+          // setDisableSubmit(false)
+          setError(
+            err === '404'
+              ? "We couldn't find that class. Please try again!"
+              : 'There was a problem joining the class, please try again!',
+          );
+          // this.setState({
+          //   disableSubmit: false,
+          //   error: (err === "404") ?
+          //     "We couldn't find that class. Please try again!" :
+          //     "There was a problem joining the class, please try again!",
+          // });
+          // console.log(err);
```

Comment:
`console.error(err)` should work? 

https://github.com/uclaacm/TeachLAFrontend/blob/master/.eslintrc.js#L122

Reply:
Yup console.error works